src/ipnet.rs

```diff
@@ -224,9 +223,11 @@ impl Ipv4Net {
         self.width == IPV4_NET_WIDTH_MAX
     }
 
-    /// Return the number of addresses contained within this subnet.
-    pub fn size(&self) -> u32 {
-        1u32 << (IPV4_NET_WIDTH_MAX - self.width)
+    /// Return the number of addresses contained within this subnet or None for
+    /// a /0 subnet whose value would be one larger than can be represented in
+    /// a `u32`.
+    pub fn size(&self) -> Option<u32> {
```

Comment:
I don't follow this. Why would a `/0` not imply a size of `u32::MAX`?

Reply:
As discussed: a /0 has 2^32 addresses in it which is one more than a u32 can represent.

Reply:
My bad here! Off by one, off by a thousand. This looks right.